crates/vm/levm/src/vm.rs

```diff
@@ -554,6 +549,29 @@ impl VM {
         // https://eips.ethereum.org/EIPS/eip-4844
         let blob_gas_cost = self.get_max_blob_gas_cost()?;
 
+        // For the transaction to be valid the sender account has to have a balance >= gas_price * gas_limit + value if tx is type 0 and 1
+        // balance >= max_fee_per_gas * gas_limit + value + blob_gas_cost if tx is type 2
+        let gas_fee_for_valid_tx = self
+            .env
+            .tx_max_fee_per_gas
+            .unwrap_or(self.env.gas_price)
+            .checked_mul(self.env.gas_limit)
+            .ok_or(VMError::TxValidation(
+                TxValidationError::GasLimitPriceProductOverflow,
+            ))?;
+
+        let balance_for_valid_tx = gas_fee_for_valid_tx
+            .checked_add(value)
+            .ok_or(InternalError::UndefinedState(1))?
+            .checked_add(blob_gas_cost)
+            .ok_or(InternalError::UndefinedState(1))?;
```

Comment:
In this case we can return an `InsufficientAccountFunds`. The user won't be able to pay for that sum anyways 😆

Reply:
Done in [b826efc](https://github.com/lambdaclass/ethrex/pull/1504/commits/b826efcbdb14308362b0e50686a7c7fa6df427a1)